assignments/this.js

```diff
@@ -1,10 +1,10 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. Window/Global Object Binding. To me, this means that "this" is always going to look outward to find meaning. It will keep searching until it hits the global object.
```

Comment:
This is actually a pretty good rule of thumb for arrow functions, but may not apply so cleanly to regular function declarations. Take a look at this:

```js
const test = {
	list: ['foo', 'bar', 'baz'],
	capitalize: function() {
		console.log(this.list.map(function (word, i) {
			const lastItem = this.list.length - 1;
			return `${word} ${this.list[lastItem - i]}`
		}))
	}
}
test.capitalize();
// output: Error: Cannot read property 'length' of undefined
```

The callback function for map is a normal anonymous function, but with those kinds of functions, the `this` gets bound to the global object, making the above fail. Using an arrow function fixes this:

```js
const test = {
	list: ['foo', 'bar', 'baz'],
	capitalize: function() {
		console.log(this.list.map((word, i) => {
			const lastItem = this.list.length - 1;
			return `${word} ${this.list[lastItem - i]}`
		}))
	}
}

test.capitalize();
// output: [ 'foo baz', 'bar bar', 'baz foo']
```

The arrow function doesn't have its own `this` binding, so will look around for a `this`, just like you described. Things like this make `this` confusing, but keep at it, and you'll soon develop this intuition.

P.S. I found this resource while googling morre about `this`: [Eyballing `this`](https://gist.github.com/zcaceres/2a4ac91f9f42ec0ef9cd0d18e4e71262)